### ISSUE TITLE
navigation menus in the header area fail SC 2.4.11/12 of WCAG 2.2

### ISSUE THREAD

**Issue by jcrawfor** [w3c/w3c-website#506]
**Describe the issue**
A clear and concise description of what the issue is.
With a menu open, the user can user can tab to interactive elements on the page that are totally obscured by the open menu.
**To reproduce**
Tab to a menu and press enter/spacebar to open, tab through the menu options and onto the next menu option. Keep tabbing until you reach links on the content area of the page that are hidden behind the menu.

**Expected behavior**
I think I would expect the menu to close when the focus moves to another element on the page. That way you never risk it hiding an element on the page that gets the focus.

**Screenshots**

**Additional context**

**Comment by japuiglaucia** [comment]
navigation menus in the header area fail SC 2.4.11/12 of WCAG 2.2 #506

**Comment by NicolaSaunders** [comment]
I would definitely appreciate clarification of how to interpret this. My current understanding, drawing from the section on [User openable, persistent disclosures](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html#user-openable-persistent-disclosures) is that the current navigation pattern is not a failure of SC 2.4.11 because the opened sub-navigation can be dismissed with the escape key. Quoting from that section:

> In recognition of more complex interfaces and user needs there is a note: Content opened by the user may obscure the component receiving focus. If the user can bring the item with focus into view using a method without having to navigate back to the user-opened content to dismiss it, this criterion would be passed. For example, keyboard actions that may allow the item with focus to be revealed include:
> - using the Escape key to dismiss the obscuring content;

**Comment by jean-gui** [comment]
@iadawn ping?

**Comment by iadawn** [comment]
So, reading through the user openable, persistent disclosures the pertinent point is:

> When focus leaves the additional content, the additional content is automatically hidden or collapsed, or the content can be hidden or collapsed by use of a dedicated keyboard command (for example, the Escape key.) This is very similar to patterns discussed previously under Non-persistent opened information. A distinguishing factor can be that the user's last point of interaction in the disclosure is preserved (it persists) even though it may be hidden until a user returns. Some trees and side navigation patterns behave this way.

In this case the menu can be collapsed using the escape key ✔️

However, when focus moves away from the menu it stays revelled. This is similar to the example [Disclosure Navigation Menu](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/) pattern. In this point 3 in the accessibility features explains the behaviour in regards to closing any open menus:

> If a dropdown is open and focus is inside the navigation region, pressing Esc will close the dropdown. Moving focus out of the navigation region also closes an open dropdown. Implementing this Esc behavior is necessary to meet the WCAG 2.1 [1.4.13: Content on Hover or Focus](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) criterion. 

The main thing to note about this example is that any open navigation menus are closed when focus moves away from the whole region. I guess this is good that if the user tabs off the bottom of one menu (except the last one) it doesn't automatically close and ruin the tab history.

However, I would probably be inclined to recommend closing the menu when focus is lost for each menu rather than for the whole region.
